### PR TITLE
Fixed metric suffix silently ignored in settings value

### PR DESCRIPTION
**Pull request by millb**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (up to few sentences, not needed for non-significant PRs):
An exception is thrown if the integral value is not parsed completely.

Detailed description (optional):
#7002

### CODE REVIEW

File: dbms/src/IO/ReadHelpers.h [hunk 1]
```diff
@@ -306,6 +306,14 @@ ReturnType readIntTextImpl(T & x, ReadBuffer & buf)
     return ReturnType(true);
 }
 
+template <typename T>
+void completeReadIntTextImpl(T & x, ReadBuffer & buf)
+{
+    readIntTextImpl<T, void>(x, buf);
+    if (!buf.eof())
```
Comment (alexey-milovidov) [inline dbms/src/IO/ReadHelpers.h]: There is `assertEOF` function.

**Comment by alexey-milovidov** [comment]
What happens if you change the semantic of `parse` function to expect eof?

**Comment by millb** [comment]
> What happens if you change the semantic of `parse` function to expect eof instead of introducing a new function?

I did this but I got a lot of errors. As I think it happens because some functions use the property of ignoring the suffix. I don't know how many such functions exist and decided that it's easier to write a new function.

**Comment by millb** [comment]
01019_Buffer_and_max_memory_usage fail isn't caused by my changes.

**Pull request merged**